tests/integrations/subprocess/test_Subprocess__stream_output.py

```diff
@@ -13,7 +21,7 @@ def popen_process():
     # continuously returning "" once it reaches EOF
     process.stdout.readline.side_effect = [
         "output line 1\n",
-        "\n",
+        sleep(0.1) or "\n",  # sneaky method to insert time passing in output
```

Comment:
So, I'm not especially proud of this (except that its clever...but that usually just gets you in trouble...).

I kept hitting race conditions during testing. The output streamer would finish too quickly and `output_streamer.is_alive()` would usually return `False`. Therefore, the inner `while` loop wouldn't run. This sleep here avoids that.